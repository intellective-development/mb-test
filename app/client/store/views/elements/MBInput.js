// @flow
import * as React from 'react';
import bindClassNames from '../../../shared/utils/bind_classnames';

import styles from './MBInput.scss';

const cn = bindClassNames(styles);

// This module exports a wrapper around the basic html input element.
// The idea is to provide a new baseline for us to work off of as we move towards component style sharing.
// Eventually, the goal is to fit this component and its props into the styleguide the design team is working towards.

type MBInputProps = {className: string, inputRef: Function} & React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>>;
export const Input = ({className, inputRef, ...input_props}: MBInputProps) => {
  const classes = cn('elMBInput', className);
  return <input {...input_props} className={classes} ref={inputRef} />;
};
