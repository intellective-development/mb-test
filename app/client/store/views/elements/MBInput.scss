@import "../style/variables";

// TODO: there are a lot of styles on the input element at the moment, we'd like to move them in here
// For now, I'm just adding in styles we should be using going forward
input.elMBInput {
  font-family: $font-family-avenir;
  border-radius: 4px;
  height: 36px; // TODO: consider just resetting this to the initial height
  margin: initial;
  // height: initial; // This enables either instances or their parents to specify the size of the input

  padding-left: 0.4em;
  padding-right: 0.4em;

  &:focus {
    outline: 0;
  }
}
