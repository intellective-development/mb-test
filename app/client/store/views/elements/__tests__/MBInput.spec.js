
import * as React from 'react';

// import TestProvider from 'client/store/__tests__/utils/TestProvider';
import * as MBInput from '../MBInput';

describe('MBInput.Input', () => {
  it('renders', () => {
    expect(shallow(<MBInput.Input />)).toMatchSnapshot();
  });
});
