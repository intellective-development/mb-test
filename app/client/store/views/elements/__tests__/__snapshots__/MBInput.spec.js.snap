// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MBInput.Input renders 1`] = `
<input
  className="elMBInput"
/>
`;
